stacks-signer/src/signerdb.rs

```diff
@@ -70,6 +83,41 @@ impl SignerDb {
         )
     }
 
+    /// Get the signer state for the provided reward cycle if it exists in the database
+    pub fn get_signer_state(&self, reward_cycle: u64) -> Result<Option<SignerState>, DBError> {
+        let result: Option<String> = query_row(
+            &self.db,
+            "SELECT state FROM signer_states WHERE reward_cycle = ?",
+            &[u64_to_sql(reward_cycle)?],
+        )?;
+
+        try_deserialize(result)
```

Comment:
Given that the callers of this function panic if it fails to load the signer state, should this function simply panic if it fails to load or deserialize state?  When does it make sense to _not_ panic?

Reply:
All functions in this module currently return errors which are panicked on by their callers. One argument for this would be that it makes testing easier, because you can assert that specific errors occur.

If we want to change this, it would be cleaner to do it as a follow-up so that it can be reviewed separately. I'd be happy to do that if you think this makes more sense.

Reply:
That answers the question -- it should return a `Result` because we can then test for specific errors, which is desirable.